core/src/main/java/org/infinispan/partitionhandling/impl/PreferConsistencyStrategy.java

```diff
@@ -179,10 +179,8 @@ public void onPartitionMerge(AvailabilityStrategyContext context,
          actualMembers.retainAll(mergedTopology.getMembers());
          mergedAvailabilityMode = AvailabilityMode.DEGRADED_MODE;
       } else {
-         log.debugf("No current topology, recovered only joiners for cache %s", context.getCacheName());
-         mergedTopology = null;
-         actualMembers = null;
-         mergedAvailabilityMode = AvailabilityMode.AVAILABLE;
+         log.debugf("No current topology, recovered only joiners for cache %s. Skipping availability update.", context.getCacheName());
+         return;
```

Comment:
I've been playing with the code a little bit....

When I reverted this change (and allowed `mergedTopology` to be `null` and `actualMembers` to be an empty list) the `ClusterStatus` created [initial topology](https://github.com/infinispan/infinispan/blob/7f991051539e0d0006ef1839c32df3fcfd5c229b/core/src/main/java/org/infinispan/topology/ClusterCacheStatus.java#L422). I assume this is not correct since we are updating [because of merge](https://github.com/infinispan/infinispan/blob/995683a3d6d27e665c186e895bdf71aa01cbb29a/core/src/main/java/org/infinispan/topology/ClusterTopologyManagerImpl.java#L499-L499). 

So unless I'm mistaken, the proposed fix seems alright. And of course, thank you for noting that the test was wrong.